interpreter/interpreter.py

```diff
@@ -31,6 +31,36 @@ def interpret_internal(ast, assignment_store):
                             pass
             case 'assign-statement':
                 assignment_store[ast[1]] = interpret_internal(ast[2], assignment_store)
+                
+            case 'assign-function':
+                parameter_list = interpret_internal(ast[2], None)
+                if parameter_list is None:
+                    parameter_list = list()
+                assignment_store[ast[1]] = {
+                    'parameter-list': parameter_list,
+                    'body': ast[3],
+                    'return-expression': ast[4]
```

Comment:
what is the purpose of the return expression here?

Reply:
Hmm, i will have to have a closer look at it. Another approach might be to propagate a return signal up the tree to the root of the function. This is how break is implemented.

Reply:
aah, okay. I was thinking of something like that, but didnt know how to do it. Will take a look at break and see if i can figure it out.